File: LMS-Backend/Models/Book.cs

namespace LMS_Backend.Models;
public class Book
{
    public int Id { get; set; }
    public string Title { get; set; } = "";
    public string Author { get; set; } = "";
    public string Description { get; set; } = "";

    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
}

File: library-frontend/src/types/Book.ts
// types/Book.ts

// Represents a Book entity returned from backend
export interface Book {
  id: number;
  title: string;
  author: string;
  description: string;
  createdAt: string; // Dates from backend usually come as ISO strings
  updatedAt: string;
}

// DTO for creating a new book (frontend -> backend)
export interface CreateBookDto {
  title: string;
  author: string;
  description: string;
}
